### ISSUE TITLE
[ENH] Facelift and add guide to contribute to Cython Documentation

### ISSUE THREAD

**Issue by gunungpw** [cython/cython#4553]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex. In my code, I would like to do [...]

Hello, new to python and programming. First, I want to say Thank you to Cython maintainer and contributor for the project. I want to contribute to Cython but I'm new to programming in general. soo I try to improve visual and guide to contribute to cython documentation. This is my first try to update sphinx v3.0 -> v4.0 and change the theme to readthedocs sphinx theme.

Here the test doc [ctx-docs](https://ctx-docs.readthedocs.io/en/latest/index.html)

I also update and reorganize the docs-requirement to match guide in `readthedocs.org` because i check many file is very old and doesn't get used in building the docs or updated like `.hgignore`. Maybe in if mantainer want PR i will try to improve guide for new contributor for writing docs like setup environment and local testing.

Sorry if my English is not too good, it's not my first language. Thank you

Best Regard,

Gunung P. W.

**Describe the solution you'd like**
A clear and concise description of what you want to happen, including code examples if applicable.

Link to test docs [ctx-docs](https://ctx-docs.readthedocs.io/en/latest/index.html)

**Describe alternatives you've considered**
-

**Additional context**
Add any other context about the feature request here.
-

**Comment by gunungpw** [comment]
# Dekstop Home Page

## Old
![Screenshot_20220103_152254](https://user-images.githubusercontent.com/55311527/147910922-66d6ff81-215d-4ef2-8513-0f397edf68f4.png)


## New
![Screenshot_20220103_152218](https://user-images.githubusercontent.com/55311527/147910888-390955a1-cb60-4fc4-ae8c-6143a8dfc9b6.png)


# Mobile

## Old
![Screenshot_20220103_151525](https://user-images.githubusercontent.com/55311527/147910425-581ed284-c2e2-4885-9d70-72689ca7e755.png)

## New
![Screenshot_20220103_151522](https://user-images.githubusercontent.com/55311527/147910509-9bf255f1-7757-44de-8f77-09727edd8135.png)

**Comment by scoder** [comment]
Thanks for coming up with this. I agree that the documentation could use a hug here and there.

Regarding the general theme changes, I'm not generally opposed. But I like green better than blue (which might just be my personal preference), and it appears that we lost the version and the donation banner along the way. Also, the logo doesn't work well on a darkish blue background – I think we need something lighter here.

**Comment by gunungpw** [comment]
Okay I will try to change the `.css` close to color scheme in official docs

**Comment by gunungpw** [comment]
for the banner, i will find a way to add it again

**Comment by da-woods** [comment]
A couple of quick comments:

1. it's obviously an improvement to have something that looks right on mobile.
2. I quite like having the whole table-of-contents available on one page like in [your first image](https://user-images.githubusercontent.com/55311527/147910922-66d6ff81-215d-4ef2-8513-0f397edf68f4.png). I think scrolling is easier than clicking, and the whole table-of-contents isn't unreasonably big

**Comment by gunungpw** [comment]
> 2. I quite like having the whole table-of-contents available on one page like in [your first image](https://user-images.githubusercontent.com/55311527/147910922-66d6ff81-215d-4ef2-8513-0f397edf68f4.png). I think scrolling is easier than clicking, and the whole table-of-contents isn't unreasonably big

Thanks, I will revert it back.

I have questions in:

- docs/src/reference/special_mention.rst
- docs/src/reference/limitations.rst
- docs/src/reference/special_mention.rst

give error in build, for not listed in `:toctree::` soo I add `:orphan:` in all of them because is the content is very old and just heading. is this correct?

**Comment by da-woods** [comment]
Yes - I think all the stuff in docs/src/reference only exists so that old links still work. It doesn't need to appear in the table of contents so adding `:orphan:` seems reasonable.
